march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -84,11 +84,13 @@ void MarchHardwareInterface::init()
   registerInterface(&effort_joint_interface_);
   registerInterface(&positionJointSoftLimitsInterface);
 
-  if (power_distribution_board_read_.getSlaveIndex() != -1)
+  hasPowerDistributionBoard = marchRobot.getPowerDistributionBoard()->getSlaveIndex() != -1;
+  if (hasPowerDistributionBoard)
   {
-    for (int i = 1; i <= 8; i++)
+    for (int i = 0; i < num_joints_; i++)
     {
-      power_distribution_board_read_.getHighVoltage().setNetOnOff(false, i);
+      int netNumber = marchRobot.getJoint(joint_names_[i]).getNetNumber();
```

Comment:
Add an error message if joint has no netnumber (aka it is -1). Just like a few lines down in this method.